rclcpp/include/rclcpp/node.hpp

```diff
@@ -1593,6 +1593,9 @@ class Node : public std::enable_shared_from_this<Node>
   const rclcpp::NodeOptions node_options_;
   const std::string sub_namespace_;
   const std::string effective_namespace_;
+
+  class NodeImpl;
+  std::shared_ptr<NodeImpl> hidden_impl_;
```

Comment:
That said, I have a slightly different comment to make.  I think we should add a small comment here as well, something along the lines of:

```suggestion
  // This member is meant to be a place to backport features into stable distributions,
  // and new features targeting Rolling should not use this.  See the comment in node.cpp for more information.
  std::shared_ptr<NodeImpl> hidden_impl_{nullptr};
```